DigitalLearningSolutions.Web/Controllers/Register/RegisterDelegateByCentreController.cs

```diff
@@ -0,0 +1,192 @@
+        }
+
+        [Route("/TrackingSystem/Delegates/Register")]
+        public IActionResult Index()
+        {
+            var centreId = User.GetCentreId();
+
+            SetCentreDelegateRegistrationData(centreId);
+
+            return RedirectToAction("PersonalInformation");
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationByCentreData>))]
+        [HttpGet]
+        public IActionResult PersonalInformation()
+        {
+            var data = TempData.Peek<DelegateRegistrationByCentreData>()!;
+
+            var model = new PersonalInformationViewModel(data);
+
+            ValidateEmailAddress(model);
+
+            return View(model);
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationByCentreData>))]
+        [HttpPost]
+        public IActionResult PersonalInformation(PersonalInformationViewModel model)
+        {
+            var data = TempData.Peek<DelegateRegistrationByCentreData>()!;
+
+            ValidateEmailAddress(model);
+
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            data.SetPersonalInformation(model);
+            TempData.Set(data);
+
+            return RedirectToAction("LearnerInformation");
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationByCentreData>))]
+        [HttpGet]
+        public IActionResult LearnerInformation()
+        {
+            var data = TempData.Peek<DelegateRegistrationByCentreData>()!;
+
+            var model = new LearnerInformationViewModel(data);
+
+            PopulateLearnerInformationExtraFields(model, data);
+
+            return View(model);
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationByCentreData>))]
+        [HttpPost]
+        public IActionResult LearnerInformation(LearnerInformationViewModel model)
+        {
+            var data = TempData.Peek<DelegateRegistrationByCentreData>()!;
+
+            var centreId = data.Centre!.Value;
+
+            customPromptHelper.ValidateCustomPrompts(
+                centreId,
+                model.Answer1,
+                model.Answer2,
+                model.Answer3,
+                model.Answer4,
+                model.Answer5,
+                model.Answer6,
+                ModelState
+            );
+
+            if (!ModelState.IsValid)
+            {
+                PopulateLearnerInformationExtraFields(model, data);
+                return View(model);
+            }
+
+            data.SetLearnerInformation(model);
+            TempData.Set(data);
+
+            return new OkResult();
+        }
+
+        private void SetCentreDelegateRegistrationData(int centreId)
+        {
+            var centreDelegateRegistrationData = new DelegateRegistrationByCentreData
+            {
+                IsCentreSpecificRegistration = true,
+                Centre = centreId
+            };

```

Comment:
Could we add a constructor that takes centreId and sets the IsCentreSpecificRegistration to true?